pipeline/main.py

```diff
@@ -1,279 +0,0 @@
-""" The main file to train a Simplified MixHopGCN model using a full graph """
-
-import argparse
-import torch
-import torch.optim as optim
-import torch.nn as nn
-import torch.nn.functional as F
-import numpy as np
-import dgl
-import dgl.function as fn
-
-from dgl.data import CoraGraphDataset, CiteseerGraphDataset, PubmedGraphDataset
-from tqdm import trange
-
-class MixHopConv(nn.Module):
-    """
-    One layer of MixHopGCN.
-    """
-    def __init__(self,
-                 in_dim,
-                 out_dim,
-                 p=[0, 1, 2],
-                 dropout=0,
-                 activation=None,
-                 batchnorm=False):
-        super(MixHopConv, self).__init__()
-        self.in_dim = in_dim
-        self.out_dim = out_dim
-        self.p = p
-        self.activation = activation
-        self.batchnorm = batchnorm
-
-        # define dropout layer
-        self.dropout = nn.Dropout(dropout)
-
-        # define batch norm layer
-        if self.batchnorm:
-            self.bn = nn.BatchNorm1d(out_dim * len(p))
-        
-        # define weight dict for each power j
-        self.weights = nn.ModuleDict({
-            str(j): nn.Linear(in_dim, out_dim) for j in p
-        })
-
-    def forward(self, graph, feats):
-        with graph.local_scope():
-            degs = graph.in_degrees().float().clamp(min=1)
-            norm = torch.pow(degs, -0.5).to(feats.device).unsqueeze(1)
-            max_j = max(self.p) + 1
-            outputs = []
-            for j in range(max_j):
-                feats = feats * norm
-                graph.ndata['h'] = feats
-                graph.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
-                feats = graph.ndata.pop('h')
-                feats = feats * norm
```

Comment:
By doing so, you assume that the graphs are undirected and `graph.in_degrees()` is the same as `graph.out_degrees()`. You may want to do a sanity check for that and leave a comment for this assumption.